include/envoy/protobuf/message_validator.h

```diff
@@ -39,6 +56,15 @@ class ValidationVisitor {
    * possible.
    **/
   virtual bool skipValidation() PURE;
+
+  /**
+   * Invoked when deprecated field is encountered.
+   * @param description human readable description of the field.
+   */
+  virtual void onDeprecatedField(absl::string_view description, bool soft_deprecation) PURE;
+
+protected:
+  void onDeprecatedFieldDefault(absl::string_view description, bool soft_deprecation);
```

Comment:
@htuch added this method as default impl to be reused by both validation visitors, but failing to compile with:
```include/envoy/protobuf/message_validator.cc:15:15: error: 'warn' was not declared in this scope
     ENVOY_LOG(warn, "Unexpected field: {}",```

Reply:
could also just duplicate this method in both derived classes